src/contracts/ProofOfPhysicalAddress.contract.js

```diff
@@ -0,0 +1,100 @@
+import Web3 from 'web3'
+import helpers from './helpers'
+
+const CONFIRM_ADDRESS_EVENT_NAME = 'LogAddressConfirmed'
+
+export default class ProofOfPhysicalAddress {
+  async init({ web3, netId, addresses }) {
+    let web3_10 = new Web3(web3.currentProvider)
+    const { PROOF_OF_PHYSICAL_ADDRESS } = addresses
+
+    const branch = helpers.getBranch(netId)
+    if (branch !== 'core') {
+      throw new Error(`ProofOfPhysicalAddress contract not deployed network "${branch}"`)
+    }
+    let proofOfPhysicalAddressAbi = await helpers.getABI(branch, 'ProofOfPhysicalAddress')
+    this.instance = new web3_10.eth.Contract(proofOfPhysicalAddressAbi, PROOF_OF_PHYSICAL_ADDRESS)
+
+    this.getUserConfirmedAddresses = this.getUserConfirmedAddresses.bind(this)
+    this.getAllEvents = this.getAllEvents.bind(this)
+    this.getConfirmedAddressesByWalletAddressAndKeccakIdentifierArray = this.getConfirmedAddressesByWalletAddressAndKeccakIdentifierArray.bind(
+      this
+    )
+  }
+
+  /**
+   * Given a walletAddress, return a promise that resolves to an array of result
+   * sets of confirmed addresses from PoBA contract.
+   * @param {String} walletAddress
+   * @return {Promise}
+   */
+  async getUserConfirmedAddresses(walletAddress) {
+    let result = []
+    try {
+      const confirmAddressEvents = await this.getAllEvents(CONFIRM_ADDRESS_EVENT_NAME)
```

Comment:
@unjapones  `getAllEvents` is seemed to be called for each validator. Maybe call it only once in `augmentValidatorsWithPoPAAddress` and pass its result into `getUserConfirmedAddresses`? It would reduce the number of requests.

Reply:
Abstracted this inside a method that get all confirmed physical addresses of a given array of wallet addresses (check last commit).